include/cpprealm/rbool.hpp

```diff
@@ -237,6 +234,27 @@ namespace realm {
         }
         return lhs.b && rhs.b;
     }
+
+    template<typename T>
+    inline rbool truepredicate(const T& o) {
+        // An empty query returns all results and one way to indicate this
+        // is to serialise TRUEPREDICATE which is functionally equivalent
+        rbool* rb = internal::get_rbool(o);
+        if (rb == nullptr)
+            throw std::runtime_error("Managed object is not used in a query context");
+        auto table = rb->q.get_table();
+        return rbool(table);
```

Comment:
Will the following series of queries be supported:

```cpp
auto res1 = realm.objects<AllTypesObject>().where([](auto &obj) {
   return obj.opt_str_col == std::nullopt;
});
auto res2 = res1.where([](auto& o) {
  return truepredicate(o);
});
```

Reply:
Yes, they are supported